crates/service-protocol/src/discovery.rs

```diff
@@ -313,8 +315,26 @@ impl ServiceDiscovery {
         build_request: impl Fn() -> Request<Body>,
         mut retry_iter: RetryIter,
     ) -> Result<(ResponseParts, Bytes), DiscoveryError> {
+        let mut try_http1 = false;
         loop {
-            let response_fut = client.call(build_request());
+            let req = build_request();
+            let req = if try_http1 {
+                let (mut parts, body) = req.into_parts();
+
+                parts.address = match parts.address {
+                    Endpoint::Http(uri, Version::HTTP_2) => Endpoint::Http(uri, Version::HTTP_11),
```

Comment:
Why aren't we always defaulting to ALPN when discovering the service endpoints?

Reply:
ALPN client will only use http2 if it detects its presence via TLS. Unencrypted, it will use http1 always. Our node services do not support http1. Therefore we must try http2 first

Reply:
Thanks for the clarification. Can you add this bit as a comment for my future self?